app/client/keybase/hashicorp/vault_test.go

```diff
@@ -0,0 +1,175 @@
+	return NewVaultKeybase(&configs.KeybaseConfig{
+		VaultAddr:      address,
+		VaultToken:     "dev-only-token",
+		VaultMountPath: "secret",
+	})
+}
+
+func TestMain(m *testing.M) {
+	// uses a sensible default on windows (tcp/http) and linux/osx (socket)
+	pool, err := dockertest.NewPool("")
+	if err != nil {
+		log.Fatalf("Could not construct pool: %s", err)
+	}
+
+	// uses pool to try to connect to Docker
+	err = pool.Client.Ping()
+	if err != nil {
+		log.Fatalf("Could not connect to Docker: %s", err)
+	}
+
+	// pulls an image, creates a container based on it and runs it
+	resource, err := pool.RunWithOptions(&dockertest.RunOptions{
+		Repository: "vault",
+		Tag:        "latest",
+		Env: []string{
+			"VAULT_DEV_ROOT_TOKEN_ID=dev-only-token",
+			"VAULT_DEV_LISTEN_ADDRESS=0.0.0.0:8200",
+		},
+		CapAdd: []string{"IPC_LOCK"},
+	}, func(config *docker.HostConfig) {
+		// set AutoRemove to true so that stopped container goes away by itself
+		config.AutoRemove = true
+		config.RestartPolicy = docker.RestartPolicy{
+			Name: "no",
+		}
+	})
+
+	if err != nil {
+		log.Fatalf("Could not start resource: %s", err)
+	}
+
+	err = resource.Expire(120) // Tell docker to hard kill the container in 2 minutes
+	if err != nil {
+		log.Fatalf("Could not set expiration on resource: %s", err)
+	}
+
+	// exponential backoff-retry, because the application in the container might not be ready to accept connections yet
+	if err = pool.Retry(func() error {
+
+		// get the port from the container
+		endpoint := resource.GetHostPort("8200/tcp")
+
+		// test connection to vault
+		vk, err = setupTestVaultKeybase(fmt.Sprintf("http://%s", endpoint))
+
+		if err != nil {
+			return err
+		}
+
+		_, err = vk.client.Sys().Health()
+
+		return err
+	}); err != nil {
+		log.Fatalf("Could not connect to docker: %s", err)
+	}
+
+	code := m.Run()
+
+	// You can't defer this because os.Exit doesn't care for defer
+	if err := pool.Purge(resource); err != nil {
+		log.Fatalf("Could not purge resource: %s", err)
+	}
+
+	os.Exit(code)
+}
+
+func TestVaultKeybase(t *testing.T) {
+
+	if vk == nil {
+		require.NoErrorf(t, fmt.Errorf("vault keybase is nil"), "vault keybase is nil")
+	}
+
+	// Test Create
+	_, err := vk.Create("passphrase", "hint")
+	require.NoErrorf(t, err, "error creating keypair: %s", err)
+
+	// Generate a new key pair
+	privKey, err := crypto.GeneratePrivateKey()
+	require.NoErrorf(t, err, "error generating key pair: %s", err)
+
+	// Test ImportFromString
+	_, err = vk.ImportFromString(privKey.String(), "passphrase", "hint")
+	require.NoErrorf(t, err, "error importing keypair: %s")
+
+	// Test GetPrivKey
+	origPrivKey := privKey.String()
+	privKey, err = vk.GetPrivKey(privKey.Address().String(), "passphrase")
+	require.NoErrorf(t, err, "error getting private key: %s")
+	assert.Equal(t, origPrivKey, privKey.String())
+
+	// Test ImportFromJSON
+	keyPair, err := crypto.CreateNewKey("passphrase2", "hint")
+	require.NoErrorf(t, err, "error creating key pair: %s")
```

Comment:
Since we're not providing any value to format `%s`, what ends up showing up?

Ditto below

Reply:
my bad, the format string is printed :sweat:

Messages:       error marshaling key pair: %s